src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs

```diff
@@ -184,405 +192,206 @@ public static partial class SymbolFinder
             if (solution.GetOriginatingProjectId(type) == null)
                 throw new ArgumentException(WorkspacesResources.Symbols_project_could_not_be_found_in_the_provided_solution, nameof(type));
 
-            return await DependentTypeFinder.FindTransitivelyDerivedClassesAsync(
-                type, solution, projects, cancellationToken).ConfigureAwait(false);
+            return await DependentTypeFinder.FindAndCacheDerivedClassesAsync(
+                type, solution, projects, transitive: true, cancellationToken).ConfigureAwait(false);
```

Comment:
to simplify the pattern, instead of DependentTypeFinder needing a 'transitive' and 'immediate' helper for every operation, it just instead takes in a boolean parameter stating what form it wants.